src/de/espend/idea/laravel/translation/TranslationReferences.java

```diff
@@ -1,129 +1,138 @@
+
+    private static MethodMatcher.CallToSignature[] TRANSLATION_KEY = new MethodMatcher.CallToSignature[] {
+        new MethodMatcher.CallToSignature("\\Illuminate\\Translation\\Translator", "get"),
+        new MethodMatcher.CallToSignature("\\Illuminate\\Translation\\Translator", "has"),
+        new MethodMatcher.CallToSignature("\\Illuminate\\Translation\\Translator", "choice"),
+        new MethodMatcher.CallToSignature("\\Illuminate\\Translation\\Translator", "transChoice"),
+    };
+
+    @Override
+    public void register(GotoCompletionRegistrarParameter registrar) {
+        registrar.register(PlatformPatterns.psiElement(), psiElement -> {
+            if(psiElement == null || !LaravelProjectComponent.isEnabled(psiElement)) {
+                return null;
+            }
+
+            PsiElement parent = psiElement.getParent();
+            if(parent != null && (
+                MethodMatcher.getMatchedSignatureWithDepth(parent, TRANSLATION_KEY) != null || PhpElementsUtil.isFunctionReference(parent, 0, "trans", "__", "trans_choice")
+            )) {
+                return new TranslationKey(parent);
+            }
+
+            // for blade @lang directive
+            if(BladePsiUtil.isDirectiveWithInstance(psiElement, "Illuminate\\Support\\Facades\\Lang", "get")) {
+                return new TranslationKey(psiElement);
+            }
+
+            return null;
+        });
+    }
+
+    @Override
+    public boolean support(@NotNull Language language) {
+        return PhpLanguage.INSTANCE == language;
+    }
+
+    public static class TranslationKey extends GotoCompletionProvider {
+
+        public TranslationKey(PsiElement element) {
+            super(element);
+        }
+
+        @NotNull
+        @Override
+        public Collection<LookupElement> getLookupElements() {
+
+            final Collection<LookupElement> lookupElements = new ArrayList<>();
+
+            CollectProjectUniqueKeys ymlProjectProcessor = new CollectProjectUniqueKeys(getProject(), TranslationKeyStubIndex.KEY);
+            FileBasedIndexImpl.getInstance().processAllKeys(TranslationKeyStubIndex.KEY, ymlProjectProcessor, getProject());
+            for(String key: ymlProjectProcessor.getResult()) {
+                lookupElements.add(LookupElementBuilder.create(key).withIcon(LaravelIcons.TRANSLATION));
+            }
+
+            return lookupElements;
+        }
+
+        @NotNull
+        @Override
+        public Collection<PsiElement> getPsiTargets(StringLiteralExpression element) {
+
+            final Set<PsiElement> priorityTargets = new LinkedHashSet<>();
+            final Set<PsiElement> targets = new LinkedHashSet<>();
+
+            final String contents = element.getContents();
+            if(StringUtils.isBlank(contents)) {
+                return targets;
+            }
+
+            final String priorityTemplate = "/" + LaravelSettings.getInstance(element.getProject()).getMainLanguage() + "/";

```

Comment:
I made it because of perfomance. It calculates once for all files which will be checked in future. If I do that, it will calculate same value for all files.